### PR TITLE
Added format as option to config

### PR DESCRIPTION
**Pull request by quintenbuis**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Fixed tickets | #376 

I've made it possible to change the format from the config. If there are other options that need to be added please leave a comment!

PS. Could the 'hacktoberfest-accepted' label be added to this commit? Thanks!

### CODE REVIEW

File: src/Application/Console/Definitions/AnalyseDefinition.php [hunk 1]
```diff
@@ -56,7 +56,7 @@ public static function get(): InputDefinition
                 null,
                 InputOption::VALUE_IS_ARRAY | InputOption::VALUE_REQUIRED,
                 'Format to output the result in [console, json, checkstyle]',
-                ['console']
+                config('insights.format', ['console'])
```
Comment (50bhan) [inline src/Application/Console/Definitions/AnalyseDefinition.php:59]: I think you can't use `config()` function here. `config()` is a helper function in `Laravel`.

File: src/Domain/Configuration.php [hunk 2]
```diff
@@ -42,6 +42,12 @@ final class Configuration
         'disable-security-check',
     ];
 
+    private const ACCEPTED_FORMATS = [
+        'console',
+        'json',
+        'checkstyle'
```
Comment (Jibbarth) [inline src/Domain/Configuration.php:48]: `github-action` format is missing. We have already a list of formatters in `src/Application/Console/Formatters/FormatResolver.php`, maybe we should use the same ?

File: src/Domain/Configuration.php [hunk 3]
```diff
@@ -126,6 +132,14 @@ public static function getAcceptedRequirements(): array
         return self::ACCEPTED_REQUIREMENTS;
     }
 
+    /**
+     * @return array<string>
+     */
+    public static function getAcceptedFormats(): array
```
Comment (Jibbarth) [inline src/Domain/Configuration.php:138]: this function is only used here, so it should be private. But regarding it's content, we could even drop it :+1:

File: src/Application/Console/Definitions/AnalyseDefinition.php (hunk 1 above)
Comment (Jibbarth) [reply src/Application/Console/Definitions/AnalyseDefinition.php:59]: Yes ! Remember PHPInsights is framework agnostic :wink:

**Review by Jibbarth — CHANGES_REQUESTED** [review]
Thanks for working on this @quintenbuis ! I left few comments, can you take them in account ?

Can you also add some tests in `tests/Domain/ConfigurationTest.php` ?

**Comment by quintenbuis** [comment]
Still need to write some tests. 🙂

**Comment by quintenbuis** [comment]
I'm rewriting the test **testItCreateAConsoleFormatterByDefault** but i'm running into an issue and can't seem to figure out how to fix it. The following way of retrieving the config parameters is causing a symfony error:
`Container::make()->get(Configuration::class)->getFormats()`
`Symfony\Component\Console\Exception\RuntimeException: The "--filter" option does not exist`

Is there another way of getting the config parameters? Or does anyone know how to fix this?

**Comment by quintenbuis** [comment]
@Jibbarth Can you help with this issue? https://github.com/nunomaduro/phpinsights/pull/441#issuecomment-720374312

**Comment by Jibbarth** [comment]
@quintenbuis Sorry for delay :sweat_smile: Did you launch your tests with a `--filter` parameters ?

I think it's because when instanciating the Container, it create an ArgvInput file that get arg from console. 

So, you can instanciate the Configuration by doing a "new Configuration(array $config)" instead of passing by Container.

Or maybe check what we do here https://github.com/nunomaduro/phpinsights/blob/master/tests/TestCase.php#L172-L176

**Comment by quintenbuis** [comment]
@Jibbarth Thanks for the tip! Will probably work on it this weekend! 🚀

**Comment by quintenbuis** [comment]
Alright i've looked into it but can't figure it out. Maybe someone else can build upon what i've created? Feel free to pick up where I left it or close this PR.

**Pull request closed without merging**